example/extensions/lib_custom_op/README.md

```diff
@@ -146,21 +155,154 @@ Let’s take a closer look at those registry functions:
+2. Run `python test_relu.py`. It’ll register the GPU `ReLU` operator in the MXNet backend, then invoke the operator by feeding an `NDArray` input with GPU context, and output the result tensor with GPU context.
+
+### Regular GPU Custom Operator
+
+Since most of the building blocks for registering GPU custom operators are the exactly same as CPU ones, the registering function for an operator supporting both GPU and CPU will look like this:
+```c++
+    REGISTER_OP(my_op_gpu)
+    ...
+    .setForward(forwardCPU, "cpu")
+    .setForward(forwardGPU, "gpu")
+    .setBackward(backwardCPU, "cpu")
+    .setBackward(backwardGPU, "gpu");
+```
+
+Note that you can write only the GPU funcitons, as operators don’t have to support both CPU and GPU (can be GPU only).
+
+After you register forward or backward functions with context name `“gpu”`, MXNet will dispatch forward or backward GPU functions you just registered when the operator is invoked with GPU context.
+
+In the registered `forwardGPU` function, specify the grid and block size and launch the CUDA kernel. In the GPU operators input and output tensors are pre-allocated on the GPU memory, just like in the CPU operator tensors are pre-allocated on the CPU memory. As a result, you don’t need to call `cudaMemcpy` to move the tensor data to GPU device.
+```c++
+    MXReturnValue forwardGPU(std::map<std::string, std::string> attrs,
+                             std::vector<MXTensor> inputs,
+                             std::vector<MXTensor> outputs,
+                             OpResource res) {
+        float* in_data = inputs[0].data<float>();
+        float* out_data = outputs[0].data<float>();
+        mx_stream_t cuda_stream = res.get_cuda_stream();
+        int64_t N = inputs[0].size();
+        int block = 256;
+        int grid = (N + (block - 1)) / block;
+        my_op_gpu_forward<<<grid,block,0,cuda_stream>>>(out_data, in_data, N);
+        return MX_SUCCESS;
+    }
+```
+
+Note that the `cuda_stream` object used for launching kernels is passed from MXNet backend via `OpResource` object. See below for details of `Operator Resource`.
+
+The CUDA kernel function should be no difference with an ordinary CUDA function.
+```c++
+    __global__ void my_op_gpu_forward(float *out, float *in, int64_t N) {
+        int tid = blockIdx.x * blockDim.x + threadIdx.x;
+        if (tid < N) {
+            // code your computational logic here
+        }
+    }
+```
+
+### Stateful GPU Custom Operator
+
+Recall that for writing custom stateful operator, you need to define a class that inherits `CustomStatefulOp` and overrides the `forward` and `backward` functions. Since stateful operator creating is context-aware, you need to create different classes for GPU and CPU stateful operators separately. the registering function will look like this:
```

Comment:
Change this:
> Recall that for writing custom stateful operator, you need

to:
Recall that for custom stateful operators you need to ...

Reply:
I think when you say it like this its confusing:
> Since stateful operator creating is context-aware, you need to create different classes for GPU and CPU stateful operators separately.

What if we said it more like:
Stateful operators are created context-aware, so you can create different classes for GPU and CPU stateful operators separately if desired.